includes/tn_upload.form.inc

```diff
@@ -59,7 +59,15 @@ function islandora_entities_get_tn_upload_form($form_state = NULL) {
  *   The drupal form state.
  */
 function islandora_entities_tn_upload_form_submit(array $form, array &$form_state) {
-  $object = $form_state['islandora']['objects'][0];
+  // Is this a thumbnail update?
+  $factory_object = islandora_object_load($form_state['islandora']['objects'][0]->id);
+  if (is_object(islandora_object_load($factory_object))) {
```

Comment:
This shouldn't do anything as passing an object to islandora_object_load is not valid, it takes a string as a parameter.